platform/view/services/comm/p2p_test.go

```diff
@@ -260,3 +274,18 @@ func SessionsForMPCTestRound(t *testing.T, bootstrapNode, node *P2PNode, bootstr
 	bootstrapNode.Stop()
 	node.Stop()
 }
+
+func Retry(attempts int, sleep time.Duration, f func() error) (err error) {
+	for i := 0; i < attempts; i++ {
+		if i > 0 {
+			time.Sleep(sleep)
+			sleep *= 2
+		}
+
+		err = f()
+		if err == nil {
+			return nil
+		}
+	}
+	return fmt.Errorf("no luck after %d attempts: last error: %v", attempts, err)
+}
```

Comment:
I suggest to put this function somewhere else, so we can easily reuse it.

Reply:
okay

Reply:
done :)